pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1777,7 +1778,16 @@ protected void handleNewTxn(CommandNewTxn command) {
             log.debug("Receive new txn request {} to transaction meta store {} from {}.",
                     requestId, tcId, remoteAddress);
         }
-        service.pulsar().getTransactionMetadataStoreService().newTransaction(tcId, command.getTxnTtlSeconds())
+        TransactionMetadataStoreService transactionMetadataStoreService =
+                service.pulsar().getTransactionMetadataStoreService();
+        if (transactionMetadataStoreService == null) {
+            ServiceUnitNotReadyException ex =
```

Comment:
may be add a ```TransactionNotAllowedException``` is better.

Reply:
We already have `NotAllowedError` in the PulsarApi.proto. We can add an exception on the broker side, but don't need to introduce a new error type in the PulsarApi.proto?

Reply:
Yes.
I don't want to add a new error in the protocol.
This is not a permission error.

I believe that 'service unit not available' is a good fit.

I reported the error on the client side, it looks to be very good

Reply:
I will change to TransactionCoordinatorNotFound
OK?